test/widgets/cards_test.dart

```diff
@@ -0,0 +1,344 @@
+        final Finder approveButton = find.byKey(const Key('approveButton'));
+        await tester.tap(approveButton);
+        //load dialog
+        await tester.pumpAndSettle();
+
+        //confirm dialog
+        final Finder confimrButton = find.byKey(const Key('approveConfirmButton'));
+        expect(confimrButton, findsOneWidget);
+        await tester.tap(confimrButton);
+
+        //load page
+        await tester.pumpAndSettle();
+
+        verifyRequestNever(processor, urlMatcher: equals('/rest/v1/rpc/approve_ride'));
+        expect(find.byKey(const Key('approveErrorSnackbar')), findsOneWidget);
+      });
+
+      testWidgets('can abort approving Ride', (WidgetTester tester) async {
+        await openApproveDialog(tester);
+
+        //abort dialog
+        final Finder cancelButton = find.byKey(const Key('approveCancelButton'));
+        expect(cancelButton, findsOneWidget);
+        await tester.tap(cancelButton);
+
+        //load page
+        await tester.pumpAndSettle();
+
+        expect(find.byType(SnackBar), findsNothing);
+        verifyRequestNever(processor, urlMatcher: equals('/rest/v1/rpc/approve_ride'));
+      });
+    });
+
+    group('reject Ride', () {
+      Future<void> openRejectDialog(WidgetTester tester) async {
+        //need scaffold for dialog
+        await pumpMaterial(tester, Scaffold(body: PendingRideCard(ride, reloadPage: () {}, drive: drive)));
+        //open dialog
+        final Finder rejectButton = find.byKey(const Key('rejectButton'));
+        await tester.tap(rejectButton);
+        //load page
+        await tester.pumpAndSettle();
+      }
+
+      testWidgets('can reject Ride', (WidgetTester tester) async {
+        await openRejectDialog(tester);
+
+        //confirm dialog
+        final Finder confimrButton = find.byKey(const Key('rejectConfirmButton'));
+        expect(confimrButton, findsOneWidget);
+        await tester.tap(confimrButton);
+
+        //load page
+        await tester.pumpAndSettle();
+
+        expect(find.byType(SnackBar), findsOneWidget);
+        verifyRequest(
+          processor,
+          urlMatcher: equals('/rest/v1/rpc/reject_ride'),
+          methodMatcher: equals('POST'),
+          bodyMatcher: equals({'ride_id': ride.id}),
+        ).called(1);
+      });
+
+      testWidgets('can abort rejecting Ride', (WidgetTester tester) async {
+        await openRejectDialog(tester);
+
+        //abort dialog
+        final Finder cancelButton = find.byKey(const Key('rejectCancelButton'));
+        expect(cancelButton, findsOneWidget);
+        await tester.tap(cancelButton);
+
+        //load page
+        await tester.pumpAndSettle();
+
+        expect(find.byKey(const Key('rejectSuccessSnackbar')), findsNothing);
+        verifyRequestNever(processor, urlMatcher: equals('/rest/v1/rpc/reject_ride'));
+      });
+    });
+  });
+
+  group('RideCard', () {
+    Future<void> loadRideCard(WidgetTester tester, List<ProfileFeature>? features) async {
```

Comment:
features should probably not be a positional parameter, it should be named

Reply:
i don't understad what you mean exactly

Reply:
i added some more xD